### PR TITLE
script: Update install procedure

### LINKED ISSUE #864: Installation strategy proposal
With the soon to be merged changes which modify `flynn-host download` to get the image manifest from the TUF repo (#617), the host and bootstrap manifests shipped with the deb package will become stale and should not be used. To fix this, we should not ship manifests directly to users and instead always consult the TUF repo.

I propose removing the OS specific packaging and just downloading and running an installer script (e.g. `curl -sL https://dl.flynn.io/installer.sh | sudo bash`).

The installer script will do the following:

* Install dependencies (e.g. `apt-get install libvirt-bin`)
* Download the latest `flynn-host` binary (e.g. `https://dl.flynn.io/flynn-host`) and verify it using a checksum embedded in the installer script
* Run `flynn-host download` to download the latest images from the TUF repo
* Save the output of `flynn-host download` (which is the manifest of downloaded images) to a local file

The following changes will then be needed:

* `flynn-host daemon` reads the manifest saved by the installer script to determine the IDs of the images it needs to use
* `flynn-host bootstrap` downloads `bootstrap_manifest.json` from the TUF repo before starting

Please add any thoughts as comments to this issue, I plan to start on this soon.

### PR DESCRIPTION
**Pull request by lmars**
Install `flynn-host` from the TUF repo using an install script which has the sha512 checksum of the binary embedded.

This makes it easier to add support for multiple Linux distros by not having to do specific packaging for those platforms.

Incremental patch for #617.

Closes #864.

### CODE REVIEW

File: script/export-components [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    exit 1
+  fi
+
+  local tuf_dir=$1
+
+  cd "${tuf_dir}"
+  tuf clean
+  cp "${ROOT}/host/bin/flynn-host"         "staged/targets/flynn-host"
+  cp "${ROOT}/host/bin/flynn-init"         "staged/targets/flynn-init"
+  cp "${ROOT}/cli/flynn-cli"               "staged/targets/flynn"
```
Comment (titanous) [inline script/export-components]: This needs an OS/architecture in the filename, as we'll want to have others in the same repo the future.

File: host/cli/download.go [hunk 2]
```diff
@@ -24,16 +24,17 @@ func init() {
...
+usage: flynn-host download [--driver=<name>] [--root=<path>] [--repository=<uri>] [--tuf-db=<path>] [--config-dir=<dir>] [--bin-dir=<dir>]
 
 Options:
   -d --driver=<name>       image storage driver [default: aufs]
   -r --root=<path>         image storage root [default: /var/lib/docker]
   -u --repository=<uri>    image repository URI [default: https://dl.flynn.io/images]
   -t --tuf-db=<path>       local TUF file [default: /etc/flynn/tuf.db]
-  -m --manifest-dir=<dir>  directory to copy manifests into [default: /etc/flynn]
+  -c --config-dir=<dir>    config directory [default: /etc/flynn]
+  -b --bin-dir=<dir>       binary directory [default: /usr/bin]
```
Comment (titanous) [inline host/cli/download.go]: Default should be `/usr/local/bin`

**Comment by lmars** [comment]
BTW I have test which exercises this workflow which I plan to add in a subsequent PR.

File: script/install-flynn.tmpl [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+USAGE
+}
+
+main() {
+  if ! is_root; then
+    fail "this script must be executed as the root user"
+  fi
+
+  check_installed "curl" "sha512sum" "apt-get" "apt-key" "initctl"
+
```
Comment (titanous) [inline script/install-flynn.tmpl]: For now we should ensure the `/etc/os-release` file says that we are running some version of Ubuntu 14.04 as well.

File: util/packer/ubuntu-14.04/scripts/install.sh [hunk 4]
```diff
@@ -153,46 +153,29 @@ add_apt_sources() {
...
+    "iptables"
     "libdevmapper-dev"
+    "libvirt-bin"
     "libvirt-dev"
     "linux-image-extra-$(uname -r)"
     "lxc-docker"
     "make"
     "mercurial"
-    "ruby2.0"
-    "ruby2.0-dev"
```
Comment (titanous) [inline util/packer/ubuntu-14.04/scripts/install.sh]: :heart_eyes:

**Comment by titanous** [comment]
This is awesome!

File: host/cli/download.go [hunk 5]
```diff
@@ -24,16 +24,17 @@ func init() {
 	}
 
 	Register("download", runDownload, `
-usage: flynn-host download [--driver=<name>] [--root=<path>] [--repository=<uri>] [--tuf-db=<path>] [--manifest-dir=<dir>]
+usage: flynn-host download [--driver=<name>] [--root=<path>] [--repository=<uri>] [--tuf-db=<path>] [--config-dir=<dir>] [--bin-dir=<dir>]
 
 Options:
   -d --driver=<name>       image storage driver [default: aufs]
   -r --root=<path>         image storage root [default: /var/lib/docker]
   -u --repository=<uri>    image repository URI [default: https://dl.flynn.io/images]
```
Comment (titanous) [inline host/cli/download.go]: Maybe this should be `https://dl.flynn.io/tuf`? `images` collides with other things like VM images

**Comment by lmars** [comment]
@titanous comments addressed

**Comment by titanous** [comment]
LGTM

**Pull request merged**